### ISSUE TITLE
Double quotes don't get converted to single quotes

### ISSUE THREAD

**Issue by tusharmath** [prettier/prettier#1947]
Version: 1.4.2

## Input

```ts
const t = ["YOU HAVEN'T JOINED ANY CONTESTS"]
```

## Output

```ts
const t = ["YOU HAVEN'T JOINED ANY CONTESTS"]
```

## Expected

```
const t = ['YOU HAVEN\'T JOINED ANY CONTESTS']
```

#### [Example]

[Example]: https://prettier.github.io/prettier/#%7B%22content%22%3A%22const%20t%20%3D%20%5B%5C%22YOU%20HAVEN'T%20JOINED%20ANY%20CONTESTS%5C%22%5D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Afalse%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D

**Comment by simonkberg** [comment]
Your expected output is invalid syntax:
```js
const t = ['YOU HAVEN'T JOINED ANY CONTESTS']
```
It seems like if the string already contains a single quote, prettier will use double quotes instead of escaping the single quote:
```js
const t = ['YOU HAVEN\'T JOINED ANY CONTESTS']
```
(it'll also [replace the above](https://prettier.github.io/prettier/#%7B%22content%22%3A%22const%20t%20%3D%20%5B'YOU%20HAVEN%5C%5C'T%20JOINED%20ANY%20CONTESTS'%5D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Afalse%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D) with the double quoted variant, where the single quote isn't escaped)

Seems reasonable to me.

**Comment by vjeux** [comment]
This is indeed the case. We should update the singleQuote option in the README to explain this if someone is up to it.

**Comment by tusharmath** [comment]
@simonkberg Updated the expected output. 

I think the problem is that tools like `tslint` and other start throwing up when they see `"` instead of `'`.

**Comment by azz** [comment]
@tusharmath You can enable `"avoid-escape"` in tslint:

```json
"quotemark": [true, "single", "avoid-escape"]
```

https://palantir.github.io/tslint/rules/quotemark/

**Comment by Gillespie59** [comment]
I think the README is already up-to-date

```
| **Quotes** - Use single quotes instead of double quotes.<br /><br />Notes:<br /> - Quotes in JSX will always be double and ignore this setting. <br /> - If the number of quotes outweighs the other quote, the quote which is less used will be used to format the string - Example: `"I'm double quoted"` results in `"I'm double quoted"` and `"This \"example\" is single quoted"` results in `'This "example" is single quoted'`. | `false` | `--single-quote` | `singleQuote: <bool>` |
```

**Comment by azz** [comment]
Yes. Fixed in #1952.
